### PR TITLE
[@types/nightwatch] fix: error using user defined globals

### LINKED ISSUE #59413: [@types/nightwatch] Missing support for accessible globals and external globals
### Discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/59412

<div type='discussions-op-text'>

<sup>Originally posted by **reallymello** March 23, 2022</sup>
In Nightwatch.js you can add name: value pairs in the globals collection through either the globals_path in config or through the globals property inside the test settings like this example below

`test_settings: {
default: {
globals: {
username: "myuser",
password: "password 
}
}
}`

and then access those values using syntax like `browser.globals.username` in the tests. However, when using the nightwatch types package in a typescript version of the test there is an error "Property 'username' does not exist on type 'NightwatchGlobals'"</div>

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

Fixes: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/59413

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
Github Actions service is facing the issue, hence GA didn't run on this PR
ref: [githubstatus.com](https://www.githubstatus.com/)

File: types/nightwatch/index.d.ts [hunk 1]
```diff
@@ -543,6 +543,7 @@ export interface NightwatchOptions {
 }
 
 export interface NightwatchGlobals {
+    [key: string]: any;
```
Comment (lloiser) [inline types/nightwatch/index.d.ts:546]: This looks ok, I just want to highlight that the `interface NightwatchGlobals` can also be extended by the user like this:
```ts
declare module "nightwatch" {
    export interface NightwatchGlobals {
        foo: string;
        bar: number;
    }
}
```

**Review by lloiser — APPROVED** [review]

File: types/nightwatch/index.d.ts (hunk 1 above)
Comment (vaibhavsingh97) [reply types/nightwatch/index.d.ts:546]: Thanks for highlighting 👍🏻, so my main reason for adding this is to reduce the problems faced by the developers (especially those who are new to typescript). This way if the choose to work with Nightwatch, then they don't face error if they try to use user-defined globals.

PS: I feel, having control over the types would be much better for bug-free codebase.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## nightwatch ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nightwatch@latest/))
was missing the following properties:
1. createClient
1. client
1. cli
1. runTests
1. CliRunner

<details>
<summary>as well as these 2 other properties...</summary>
<p>initClient, runner</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ad8f3f983eec2688108525993b9e38ca64584dee
</p>

File: types/nightwatch/index.d.ts [hunk 2]
```diff
@@ -1640,7 +1641,7 @@ export interface EnhancedPageObjectSections {
 export type EnhancedPageObject<
     Commands = {},
     Elements = {},
-    Sections extends EnhancedPageObjectSections = {},
```
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply types/nightwatch/index.d.ts]: This is fixed

**Review by swrdfish — APPROVED** [review]

**Comment by swrdfish** [comment]
Ready to merge

**Comment by swrdfish** [comment]
Ready to merge

**Pull request merged**